src/systemctl/systemctl.c

```diff
@@ -7336,6 +7341,53 @@ static int unit_file_create_new(
         if (r < 0)
                 return r;
 
+        if (unit_paths) {
+                _cleanup_free_ char *tmp_contents = NULL;
+                _cleanup_fclose_ FILE *f = NULL;
+                char **path;
+                size_t size;
+
+                r = read_full_file(tmp_path, &tmp_contents, &size);
+                if (r < 0)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", tmp_path);
+
+                f = fopen(tmp_path, "we");
+                if (!f)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", tmp_path);
+
+                fprintf(f,
+                        "### Editing %s\n"
+                        EDIT_MARKER_START
+                        "\n\n%s\n\n"
+                        EDIT_MARKER_END
+                        "\n",
+                        new_path,
+                        tmp_contents);
```

Comment:
tiny tweak. maybe insert the "\n" after `tmp_contents` only if it doesn't end in a trailing newline anyway...

Hmm, so create_edit_temp_file() above created tmp_path via copy_file() based on the original file, and you read that back here and write it out again, replacing it. I think it would be better if create_edit_temp_file() would insert the header/footer right away if needed, so that the file is correctly on order right-away.

Reply:
Added a strv arg to `create_edit_temp_file`. If `original_path` is specified it is simply copied, otherwise the header & footer are inserted using the files in `original_unit_paths`